spyder/widgets/fileswitcher.py

```diff
@@ -437,7 +442,12 @@ def get_editor(self, index=None, path=None):
         elif path:
             return self.tabs.widget(index)
         else:
-            return self.parent().get_current_editor()
+            # this try is needed in order to get
+            # the current client in the notebookplugin
+            try:
+                return self.parent().get_current_editor()
+            except AttributeError:
+                return self.parent().get_current_client()
```

Comment:
I don't like this because it would only work for the Notebook plugin.

Please change this whole block to

```python
self.parent().tabs.currentWidget()
```

because that's what `get_current_editor` returns, and in `NotebookPlugin` rename `self.tabwidget` to `self.tabs`.

That should work for both the Editor and the Notebook without specializing for any particular plugin.